weldx/measurement.py

```diff
@@ -47,55 +55,459 @@ class DataTransformation:
 
 
 @dataclass
-class Source:
+class SignalTransformation:
+    name: str
+    error: Error
+    input_signal: Signal
+    output_signal: Signal
+    func: "MathematicalExpression" = None
+    input_shape: Tuple = None
+    output_shape: Tuple = None
+
+
+@dataclass
+class SignalSource:
     """Simple dataclass implementation for signal sources."""
 
     name: str
     output_signal: Signal
     error: Error
 
 
+# equipment ----------------------------------------------------------------------------
 @dataclass
+class GenericEquipment:
+    """Simple dataclass implementation for generic equipment."""
+
+    name: str
+    sources: List = field(default_factory=lambda: [])
+    data_transformations: List = field(default_factory=lambda: [])
```

Comment:
maybe just `transformations` or `signal_transformations` is more fitting here?

Reply:
Just didn't bother with renaming stuff so far. But renaming it to transformations is already on my list.